tfx/components/example_gen/utils.py

```diff
@@ -210,83 +214,171 @@ def _glob_to_regex(glob_pattern: Text) -> Text:
+  """
+
   split_pattern = os.path.join(uri, split.pattern)
-  if split_pattern.count(SPAN_SPEC) != 1:
-    raise ValueError('Only one {SPAN} is allowed in %s' % split_pattern)
-
-  split_glob_pattern = split_pattern.replace(SPAN_SPEC, '*')
-  logging.info('Glob pattern for split %s: %s', split.name, split_glob_pattern)
-  split_regex_pattern = _glob_to_regex(split_pattern).replace(SPAN_SPEC, '(.*)')
-  logging.info('Regex pattern for split %s: %s', split.name,
-               split_regex_pattern)
-  if re.compile(split_regex_pattern).groups != 1:
-    raise ValueError('Regex should have only one group')
-
-  files = tf.io.gfile.glob(split_glob_pattern)
+
+  split_glob_pattern = split_pattern
+  split_regex_pattern = _glob_to_regex(split_pattern)
+  
   latest_span = None
-  for file_path in files:
-    result = re.search(split_regex_pattern, file_path)
-    if result is None:
-      raise ValueError('Glob pattern does not match regex pattern')
-    try:
-      span = int(result.group(1))
-    except ValueError:
-      raise ValueError('Cannot not find span number from %s based on %s' %
-                       (file_path, split_regex_pattern))
-    if latest_span is None or span >= int(latest_span):
-      # Uses str instead of int because of zero padding digits.
-      latest_span = result.group(1)
-
-  if latest_span is None:
-    raise ValueError('Cannot not find matching for split %s based on %s' %
-                     (split.name, split.pattern))
-  return latest_span
-
-
-def calculate_splits_fingerprint_and_span(
+  latest_version = None
+
+  if SPAN_SPEC in split.pattern:
+    if split.pattern.count(SPAN_SPEC) != 1:
+      raise ValueError('Only one %s is allowed in %s' % (SPAN_SPEC,
+                                                         split.pattern))
+
+    split_glob_pattern = split_glob_pattern.replace(SPAN_SPEC, '*')
+    split_regex_pattern = split_regex_pattern.replace(SPAN_SPEC,
+        '(?P<{}>.*)'.format(SPAN_PROPERTY_NAME))
+
+    is_match_version = VERSION_SPEC in split.pattern
+    if is_match_version and split.pattern.count(VERSION_SPEC) != 1:
+      raise ValueError('Only one %s is allowed in %s' % (VERSION_SPEC,
+                                                           split.pattern))
+    if is_match_version:
+      split_glob_pattern = split_glob_pattern.replace(VERSION_SPEC, '*')
+      split_regex_pattern = split_regex_pattern.replace(VERSION_SPEC,
+          '(?P<{}>.*)'.format(VERSION_PROPERTY_NAME))
+    
+    logging.info('Glob pattern for split %s: %s', split.name,
+                 split_glob_pattern)
+    logging.info('Regex pattern for split %s: %s', split.name,
+                 split_regex_pattern)
+
+    num_groups = re.compile(split_regex_pattern).groups
+    if is_match_version and num_groups != 2:
+      raise ValueError('Span and version regex should have two groups')
+    elif not is_match_version and num_groups != 1:
+      raise ValueError('Span only regex should have one group')
```

Comment:
this seems duplicate with about split.pattern.count check?

(if you want to double check, you can do `assert` cause it seems a bug at our side if this triggers, not sth users need to fix)

Reply:
yep its remainder from the old glob function, checked it with assert, ill fix that